### PR TITLE
Dream Logic - Merge Me!

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Makes candles be able to be endless via magic
Adds Occult things
- Note this was legit 1:1 with a dream I had last night and then expanded a bit to be balanced

### CODE REVIEW

**Comment by r4d6** [comment]
I'll look over it when I get home.

File: code/game/objects/effects/decals/crayon.dm [hunk 1]
```diff
@@ -3,29 +3,205 @@
...
+			if(M.maxHealth <= 30)
+				to_chat(M, "<span class='info'>You try to do as the book wrights but the failing body prevents you form even mubbling a single more word.</span>")
+				return
+
+			var/obj/item/flame/candle/mage_candle
+			var/datum/reagent/organic/blood/B = M.get_blood()
+			var/obj/effect/decal/cleanable/blood/writing/spell
+			var/candle_amount = 0
+			to_chat(M, "<span class='info'>The wrightings inside the book and the rune drawn match up.</span>")
+			for(mage_candle in oview(3))
```
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:50]:
```suggestion
			for(var/obj/item/flame/candle/mage_candle in oview(3))
```

File: code/game/objects/effects/decals/crayon.dm [hunk 2]
```diff
@@ -3,29 +3,205 @@
...
+				if(!mage_candle.lit)
+					mage_candle.light(flavor_text = SPAN_NOTICE("\The [name] lights up."))
+					mage_candle.endless_burn = TRUE
+					B.remove_self(15)
+					to_chat(M, "<span class='info'>Something lights a candle.</span>")
+				candle_amount += 1
+			for(spell in oview(3)) //Dont forget to clear your old work
```
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:57]:
```suggestion
			for(var/obj/effect/decal/cleanable/blood/writing/spell in oview(3)) //Dont forget to clear your old work
```

File: code/game/objects/effects/decals/crayon.dm [hunk 3]
```diff
@@ -3,29 +3,205 @@
...
+		if(M.disabilities&NEARSIGHTED && is_rune && M.species?.reagent_tag != IS_SYNTHETIC)
+
+			//Anti-Death check
+			if(M.maxHealth <= 30)
+				to_chat(M, "<span class='info'>You try to do as the book wrights but the failing body prevents you form even mubbling a single more word.</span>")
+				return
+
+			var/obj/item/flame/candle/mage_candle
+			var/datum/reagent/organic/blood/B = M.get_blood()
+			var/obj/effect/decal/cleanable/blood/writing/spell
```
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:47]:
```suggestion
```

File: code/game/objects/effects/decals/crayon.dm [hunk 4]
```diff
@@ -3,29 +3,205 @@
...
+	..()
+	if(istype(I, /obj/item/oddity/common/book_unholy) || istype(I, /obj/item/oddity/common/book_omega))
+		if(M.disabilities&NEARSIGHTED && is_rune && M.species?.reagent_tag != IS_SYNTHETIC)
+
+			//Anti-Death check
+			if(M.maxHealth <= 30)
+				to_chat(M, "<span class='info'>You try to do as the book wrights but the failing body prevents you form even mubbling a single more word.</span>")
+				return
+
+			var/obj/item/flame/candle/mage_candle
```
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:45]:
```suggestion
```

File: code/game/objects/effects/decals/crayon.dm [hunk 5]
```diff
@@ -3,29 +3,205 @@
...
+							lesser.colony_friend = TRUE
+							lesser.friendly_to_colony = TRUE
+							lesser.faction = "Living Dead"
+							lesser.maxHealth = lesser.maxHealth * 0.5
+							lesser.health =  lesser.maxHealth * 0.5
+							M.maxHealth -= 30
+							M.health -= 30
+							B.remove_self(50)
+							return
+						return
```
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:123]:
```suggestion
					for(var/mob/living/carbon/superior_animal/greater in  oview(1) && M.maxHealth >= 30)
						to_chat(M, "<span class='info'>To raise the dead one must use self, and in self we draw closer to death...</span>")
						greater.revive()
						greater.colony_friend = TRUE
						greater.friendly_to_colony = TRUE
						greater.friends += M
						greater.faction = "Living Dead"
						M.maxHealth -= 30
						M.health -= 30
						B.remove_self(70)
					for(var/mob/living/simple_animal/lesser in  oview(1) && M.maxHealth >= 30)
						to_chat(M, "<span class='info'>To raise the dead one must use self, and in self we draw closer to death...</span>")
						lesser.revive()
						lesser.colony_friend = TRUE
						lesser.friendly_to_colony = TRUE
						lesser.faction = "Living Dead"
						M.maxHealth -= 30
						M.maxHealth -= 25
						M.health -= 25
						B.remove_self(50)
```

File: code/game/objects/effects/decals/crayon.dm (hunk 3 above)
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:47]: Why is spell & mage_candle defined here if they are only used for the for() loop.

File: code/game/objects/effects/decals/crayon.dm [hunk 6]
```diff
@@ -3,29 +3,205 @@
...
+			if(L.name == LANGUAGE_CULT)
+				able_to_cast = TRUE // We can cast
+			if(L.name == LANGUAGE_OCCULT)
+				able_to_cast = TRUE // We can cast
+
+			var/obj/item/flame/candle/mage_candle
+			var/datum/reagent/organic/blood/B = M.get_blood()
+			var/obj/effect/decal/cleanable/blood/writing/spell
+			var/candle_amount = 0
+			for(mage_candle in oview(3))
```
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:170]:
```suggestion
			var/datum/reagent/organic/blood/B = M.get_blood()
			var/candle_amount = 0
			var/obj/item/flame/candle/for(mage_candle in oview(3))
```

File: code/game/objects/effects/decals/crayon.dm [hunk 7]
```diff
@@ -3,29 +3,205 @@
...
+				if(!mage_candle.lit)
+					mage_candle.light(flavor_text = SPAN_NOTICE("\The [name] lights up."))
+					mage_candle.endless_burn = TRUE
+					B.remove_self(15)
+					to_chat(M, "<span class='info'>Something lights a candle.</span>")
+				candle_amount += 1
+
+			for(spell in oview(3)) //Dont forget to clear your old work
```
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:178]:
```suggestion
			for(var/obj/effect/decal/cleanable/blood/writing/spell in oview(3)) //Dont forget to clear your old work
```

File: code/game/objects/effects/decals/crayon.dm [hunk 8]
```diff
@@ -3,29 +3,205 @@
...
+					for(greater in  oview(1) && M.maxHealth >= 30)
+						to_chat(M, "<span class='info'>The body before you is no more as its code is one with you.</span>")
+						if(able_to_cast)
+							M.maxHealth += 1
+							M.health += 1
+							M.unnatural_mutations.total_instability += 1 //A soft cap
+						B.remove_self(70)
+						greater.dust()
+						return
+					for(lesser in  oview(1) && M.maxHealth >= 30)
```
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:198]:
```suggestion
					for(var/mob/living/carbon/superior_animal/greater in  oview(1) && M.maxHealth >= 30)
						to_chat(M, "<span class='info'>The body before you is no more as its code is one with you.</span>")
						if(able_to_cast)
							M.maxHealth += 1
							M.health += 1
							M.unnatural_mutations.total_instability += 1 //A soft cap
						B.remove_self(70)
						greater.dust()
						return
					for(var/mob/living/simple_animal/lesser in  oview(1) && M.maxHealth >= 30)
```

**Review by r4d6 — COMMENTED** [review]
Small suggestions here and there, mostly about the `for()` loops.

File: code/game/objects/effects/decals/crayon.dm [hunk 9]
```diff
@@ -3,29 +3,228 @@
...
+			if(M.maxHealth <= 30)
+				to_chat(M, "<span class='info'>Your hand shakes to much as your body fails you to do anything of value here.</span>")
+				return
+		to_chat(M, "<span class='info'>The writings of the rune are correct.</span>")
+		var/able_to_cast = FALSE
+		for(var/datum/language/L in M.languages)
+			if(L.name == LANGUAGE_CULT)
+				able_to_cast = TRUE // We can cast
+			if(L.name == LANGUAGE_OCCULT)
+				able_to_cast = TRUE // We can cast
```
Comment (r4d6) [inline code/game/objects/effects/decals/crayon.dm:98]:
```suggestion
		for(var/datum/language/L in M.languages)
			if(L.name == LANGUAGE_CULT || L.name == LANGUAGE_OCCULT)
				able_to_cast = TRUE // We can cast
```

**Comment by Yawet330** [comment]
incredibly based

**Pull request merged**